wordpress-mobile/WordPress-Android
# Fix failing connected tests

Fixes connected tests that were failing for `loginWithMagicLink` due to non-null parameter being null.

I believe this was introduced by [this change](https://github.com/wordpress-mobile/WordPress-Android/commit/c9854692e53e302b4ae3bc42a4ff5af35e519065#diff-1cf6428dcfe0d429dfdcf7398ef4c056R812) from [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12728).

This PR simply updates `onResume(site: SiteModel)` to `onResume(site: SiteModel?)` as it seems to have fixed the issue, but I wonder if we shouldn't do the same for the other functions in that class that also receive a `SiteModel`? (cc @aforcier as I don't have much context on this right now)

To test:

Run the connected tests from the `LoginTests` suite and make sure they all pass.

PR submission checklist:
